numpy/core/arrayprint.py

```diff
@@ -442,7 +444,7 @@ def _array2string(a, options, separator=' ', prefix=""):
     # skip over array(
     next_line_prefix += " "*len(prefix)
 
-    lst = _formatArray(a, format_function, a.ndim, options['linewidth'],
+    lst = _formatArray(a, format_function, options['linewidth'],
```

Comment:
Would things break if you called this on `data` instead of on `a`? If not, then the changes below to `_formatArray` are also unnecessary, though, like for `_leading_trailing`, prettier or more performant.

Reply:
It depends what you mean by break. Calling this on `MaskedArray` currently warns and converts `masked` to `nan` / errors on `int`. With your proposed change, it would silently discard the mask.

Ideally, we'd allow the formatters to see the subclasses if they want to, but that's out of scope for 1.14